src/Chapter3.hs

```diff
@@ -479,6 +501,32 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living people inside
 -}
+data City = City
+    { cityCastle            :: Castle
+    , cityCommunityBuilding :: CommunityBuilding
+    , cityHouses            :: [Int]
+    }
+
+data Castle
+    = None
+    | CastleWithWalls
+    | CastleWithoutWalls
```

Comment:
Nice 👍🏼 

In order to make it more interesting, you can try to have the castle with the name (as in the task). Let's see which functions it would affect 🙂